vimscript.html.markdown

```diff
@@ -0,0 +1,572 @@
+let [mother, father; children] = ['Alice', 'Bob', 'Carol', 'Dennis', 'Emily']
+```
+
+## Flow control
+
+### Conditional
+
+Conditions are set between `if` and `endif`. They can be nested.
+
+```vim
+if condition
+	  echo 'First condition'
+elseif another_condition
+	  echo 'Second condition'
+else
+	  echo 'Fail'
+endif
+```
+
+### Loops
+
+Two types of loops: `:for` and `:while`. Use `:continue` to skip to the next
+iteration, `:break` to break out of the loop.
+
+#### For-loop
+
+For-loops iterate over lists and nothing else. If you want to iterate over
+another sequence you need to use a function which will create a list.
+
+```vim
+" Iterate over a list
+for person in ['Alice', 'Bob', 'Carol', 'Dennis', 'Emily']
+		echo 'Hello ' . person
+endfor
+
+" Iterate over a nested list by unpacking it
+for [x, y] in [[1, 0], [0, 1], [-1, 0], [0, -1]]
+		echo 'Position: x ='  . x . ', y = ' . y
+endfor
+
+" Iterate over a range of numbers
+for i in range(10, 0, -1)  " Count down from 10
+    echo 'T minus'  . i
+endfor
+
+" Iterate over the keys of a dictionary
+for symbol in keys({'π': 3.14, 'e': 2.71})
+		echo 'The constant ' . symbol . ' is a transcendent number'
+endfor
+
+" Iterate over the values of a dictionary
+for value in keys({'π': 3.14, 'e': 2.71})
+		echo 'The value ' . symbol . ' approximates a transcendent number'
+endfor
+
+" Iterate over the keys and values of a dictionary
+for [symbol, value] in items({'π': 3.14, 'e': 2.71})
+		echo 'The number ' . symbol . ' is approximately ' . value
+endfor
+```
+
+#### While-loops
+
+```vim
+while !there_yet
+		echo 'Are we there yet?'
+endwhile
+```
+
+### Exception handling
+See `:h exception-handling`
+
+Throw new exceptions as strings, catch them by pattern-matching a regular
+expression against the string
+
+```vim
+" Throw new exception
+throw "Wrong arguments"
+
+" Guard against an exception (the second catch matches any exception)
+try
+		source path/to/file
+catch /Cannot open/
+		echo 'Looks like that file does not exist'
+catch /.*/
+    echo 'Something went wrong, but I don't know what'
+finally
+		echo 'I'm done trying'
+endtry
+```
+
+
+## Functions
+
+### Defining functions
+
+```vim
+" Function names have to start with a capital letter
+function! AddNumbersLoudly(x, y)
+    " Use a: scope to access arguments
+    echo 'Adding'  . a:x .  'and'  . a:y
+    return a:x + a:y
+endfunction
+
+" Can start with lower-case letter if local to script
+function! s:addNumbersLoudly(x, y)
+    echo 'Adding'  . a:x .  'and'  . a:y
+    return a:x + a:y
+endfunction
+
+" Closures, functions carrying values from outer scope (:h :func-closure)
+function! MakeAdder(x)
+    function! Adder(n) closure
+        return n + x
+    endfunction
+    return funcref('Adder')
+endfunction
+let AddFive = MakeAdder(5)
+echo AddFive(3)               |" Prints 8
+
+" Aborting functions, abort once error occurs (:h :func-abort)
+function! SourceMyFile() abort
```

Comment:
Maybe add comment stating the `abort` was added after the function name since a reader may miss this.

Reply:
I have split the code block and added a sentence for all the special annotations. That should make it clear.